routers/web/org/times.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package org
+
+import (
+	"net/http"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/builder"
+)
+
+const (
+	tplByRepos      base.TplName = "org/times/times_by_repos"
+	tplByMembers    base.TplName = "org/times/times_by_members"
+	tplByMilestones base.TplName = "org/times/times_by_milestones"
+)
+
+// resultTimesByRepos is a struct for DB query results
+type resultTimesByRepos struct {
+	Name    string `xorm:"name"`
+	SumTime int64  `xorm:"sumtime"`
+}
+
+// resultTimesByMilestones is a struct for DB query results
+type resultTimesByMilestones struct {
+	RepoName     string `xorm:"reponame"`
+	Name         string `xorm:"name"`
+	ID           string `xorm:"id"`
+	SumTime      int64  `xorm:"sumtime"`
+	HideRepoName bool   `xorm:"-"`
+}
+
+// resultTimesByMembers is a struct for DB query results
+type resultTimesByMembers struct {
+	Name    string `xorm:"name"`
+	SumTime int64  `xorm:"sumtime"`
+}
+
+// parseTimes contains functionality that is required in all these functions,
+// like parsing the date from the request, setting default dates, etc.
+func parseTimes(ctx *context.Context) (unixfrom, unixto int64, err error) {
+	// View variables
+	ctx.Data["PageIsOrgTimes"] = true
+	ctx.Data["AppSubURL"] = setting.AppSubURL
+
+	// Time range from request, if any
+	from := ctx.FormString("from")
+	to := ctx.FormString("to")
+	// Defaults for "from" and "to" dates, if not in request
+	if from == "" {
+		// DEFAULT of "from": start of current month
+		from = time.Now().Format("2006-01") + "-01"
+	}
+	if to == "" {
+		// DEFAULT of "to": today
+		to = time.Now().Format("2006-01-02")
+	}
+
+	// Prepare Form values
+	ctx.Data["RangeFrom"] = from
+	ctx.Data["RangeTo"] = to
+
+	// Prepare unix time values for SQL
+	gfrom, err := time.Parse("2006-01-02", from)
```

Comment:
what does the `g` represent in the variable?

Reply:
It stands for "Go", as in Go's Time object. The variable named `from` above is of String type, and here we convert to Go's Time. It's a temporary variable that is needed for the handling of `err` from the `Parse` function. Changed it to `from2`.